lib/core/src/Cardano/Wallet/DB/Model.hs

```diff
@@ -168,6 +171,12 @@ mReadCheckpoint :: Ord wid => wid -> ModelOp wid s t xprv (Maybe (Wallet s t))
 mReadCheckpoint wid db@(Database wallets _) =
     (Right (checkpoints <$> Map.lookup wid wallets >>= listToMaybe), db)
 
+mRollbackTo :: Ord wid => wid -> SlotId -> ModelOp wid s t xprv ()
+mRollbackTo wid pt = alterModel wid $ \wal ->
+    ((), wal { checkpoints = filter keepBeforePoint (checkpoints wal) })
+  where
+    keepBeforePoint cp = (slotId :: BlockHeader -> SlotId) (currentTip cp) < pt
```

Comment:
I guess we also need to recover pending tx data here.

Reply:
Good point. I am only seeing your comment now :upside_down_face: ... I am heading off for today. This echoes the second "TODO" I've written above; testing only looks at the checkpoints so far. I'd suggest to keep the Model that way, write a test about the history on rollbacks, see it fails and only fix it after that (another PR is fine for this I think).